pkg/proxy/logger/endpoint/epinfo.go

```diff
@@ -73,7 +74,15 @@ func (r *endpointInfoRegistry) FillEndpointInfo(info *accesslog.EndpointInfo, ip
 		}
 		// Default to WORLD if still unknown
 		if id == 0 {
-			id = identity.ReservedIdentityWorld
+			if option.Config.IsDualStack() {
+				if ipAddr.To4() == nil {
+					id = identity.ReservedIdentityWorldIPv6
+				} else {
+					id = identity.ReservedIdentityWorldIPv4
+				}
+			} else {
+				id = identity.ReservedIdentityWorld
+			}
```

Comment:
Maybe it's best to define a util function somewhere that returns the identity based on the IP, so that we keep the code DRY and prevent bugs such as [1]